### ISSUE TITLE
[CI-fail] Viewbox Shape Mismatch

### ISSUE THREAD

**Issue by j9ac9k** [pyqtgraph/pyqtgraph#896]
Overwhelming majority of CI failures on non-Linux systems are viewbox dimensions related issues

```
=================================== FAILURES ===================================
_________________________________ test_ViewBox _________________________________

    def test_ViewBox():
        init_viewbox()
    
        w = vb.geometry().width()
        h = vb.geometry().height()
        view1 = QRectF(0, 0, 10, 10)
        size1 = QRectF(0, h, w, -h)
        assertMapping(vb, view1, size1)
    
        # test resize
        win.resize(400, 400)
        app.processEvents()
        w = vb.geometry().width()
        h = vb.geometry().height()
        size1 = QRectF(0, h, w, -h)
        assertMapping(vb, view1, size1)
    
        # now lock aspect
        vb.setAspectLocked()
    
        # test wide resize
        win.resize(800, 400)
        app.processEvents()
        w = vb.geometry().width()
        h = vb.geometry().height()
        view1 = QRectF(-5, 0, 20, 10)
        size1 = QRectF(0, h, w, -h)
        assertMapping(vb, view1, size1)
    
        # test tall resize
        win.resize(400, 800)
        app.processEvents()
        w = vb.geometry().width()
        h = vb.geometry().height()
        view1 = QRectF(0, -5, 10, 20)
        size1 = QRectF(0, h, w, -h)
>       assertMapping(vb, view1, size1)

pyqtgraph/graphicsItems/ViewBox/tests/test_ViewBox.py:73: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

vb = <pyqtgraph.graphicsItems.ViewBox.ViewBox.ViewBox(0x7fb31260c160, parent=0x7fb3148d02d0, pos=0,0, z=-100, flags=(ItemIsFocusable|ItemClipsChildrenToShape|ItemUsesExtendedStyleOption|ItemSendsGeometryChanges))  at 0x1229feec8>
r1 = PySide2.QtCore.QRectF(0.000000, -5.000000, 10.000000, 20.000000)
r2 = PySide2.QtCore.QRectF(0.000000, 771.000000, 400.000000, -771.000000)

    def assertMapping(vb, r1, r2):
>       assert vb.mapFromView(r1.topLeft()) == r2.topLeft()
E       assert PySide2.QtCor...0, 785.500000) == PySide2.QtCore...0, 771.000000)
E         -PySide2.QtCore.QPointF(0.000000, 785.500000)
E         +PySide2.QtCore.QPointF(0.000000, 771.000000)

pyqtgraph/graphicsItems/ViewBox/tests/test_ViewBox.py:10: AssertionError
```

Fixing this issue would resolve almost the remainder of the CI errors.

**Comment by Dschoni** [comment]
On Windows 10, I seem to be able to reproduce locally. With the following environment

```
atomicwrites              1.3.0                    py27_1
attrs                     19.1.0                   py27_1
backports                 1.0                      py27_1
blas                      1.0                         mkl
certifi                   2019.3.9                 py27_0
colorama                  0.4.1                    py27_0
configparser              3.7.4                    py27_0
coverage                  4.5.3            py27h0c8e037_0
entrypoints               0.3                      py27_0
enum34                    1.1.6                    py27_1
flake8                    3.7.7                    py27_0
funcsigs                  1.0.2            py27h8885ae1_0
functools32               3.2.3.2                  py27_1
icc_rt                    2019.0.0             h0cc432a_1
intel-openmp              2019.3                      203
jpeg                      8d                        vc9_2
libpng                    1.6.37               h7a46e7a_0
libtiff                   4.0.6                     vc9_2
mccabe                    0.6.1                    py27_1
mkl                       2019.3                      203
mkl_fft                   1.0.12           py27h44c1dab_0
more-itertools            5.0.0                    py27_0
numpy                     1.16.3           py27h5fc8d92_0
numpy-base                1.16.3           py27hb1d0314_0
pathlib2                  2.3.3                    py27_0
pip                       19.1.1                   py27_0
pluggy                    0.11.0                     py_0
py                        1.8.0                    py27_0
pycodestyle               2.5.0                    py27_0
pyflakes                  2.1.1                    py27_0
pyopengl                  3.1.1a1                  py27_0
pyqt                      4.11.4                   py27_7
pyqtgraph                 0.11.0.dev0+g20c7495          pypi_0    pypi
pytest                    4.5.0                    py27_0
python                    2.7.16               hcb6e200_0
qt                        4.8.7                     vc9_6
scandir                   1.10.0           py27h0c8e037_0
scipy                     1.2.1            py27h4c3ab11_0
setuptools                41.0.1                   py27_0
sip                       4.18                     py27_0
six                       1.12.0                   py27_0
sqlite                    3.28.0               h0c8e037_0
typing                    3.6.6                    py27_0
vc                        9                    h7299396_1
vs2008_runtime            9.00.30729.1         hfaea7d5_1
wcwidth                   0.1.7            py27hb1a0d82_0
wheel                     0.33.4                   py27_0
wincertstore              0.2              py27hf04cefb_0
zlib                      1.2.11               h3cc03e0_3
```
Trying to dig into the error.

**Comment by Dschoni** [comment]
Sorry no. My error comes from the (right now uncommented) line 92. So in fact, can't reproduce this exact error locally.

**Comment by j9ac9k** [comment]
Fixed with #978 being merged
